### DISCUSSION TITLE
Enhancement: all i want all i need please all i need to be a god at this game is QWER controls  wasd just dont fit

### DISCUSSION THREAD

**Discussion by MajinCanon** [FunkinCrew/Funkin#1610] [Ideas]

**Comment by MarioLuigi124** [comment]
use the arrow keys...

**Comment by TentaRJ** [comment]
Next time, please use the template next time you create an issue.

And yeah just use arrows.

**Comment by foxsouns** [comment]
i think they want a layout with a 1:1 of the game, instead of the way its set up now. like, on the qwerty keyboards, theyre asking for qwer, so that it matches up with the game better?

**Comment by TentaRJ** [comment]
Inside Controls.hx, you can change button placement.
```haxe
switch (scheme)
		{
			case Solo:
				bindKeys(Control.UP, [E, FlxKey.UP]);
				bindKeys(Control.DOWN, [W, FlxKey.DOWN]);
				bindKeys(Control.LEFT, [Q, FlxKey.LEFT]);
				bindKeys(Control.RIGHT, [R, FlxKey.RIGHT]);
				bindKeys(Control.ACCEPT, [Z, SPACE, ENTER]);
				bindKeys(Control.BACK, [BACKSPACE, ESCAPE]);
				bindKeys(Control.PAUSE, [P, ENTER, ESCAPE]);
				bindKeys(Control.RESET, [T]);
		}
```

**Comment by exurd** [comment]
> all i want all i need please all i need to be a god at this game is QWER controls wasd just dont fit

I think it's better to use "UIOP" in case anyone wants to use the WASD keys instead of the arrow keys.

> Inside Controls.hx, you can change button placement.

 The better solution for this issue is to just add control settings in-game that allow you to bind any keys you want. That way, you can use any layout you want.

**Comment by GStudiosX2** [comment]
It does support Wasd but I think if u don't have a qwerty keyboard it might not work

**Comment by isakube** [comment]
> Inside Controls.hx, you can change button placement.
> 
> ```hx
> switch (scheme)
> 		{
> 			case Solo:
> 				bindKeys(Control.UP, [E, FlxKey.UP]);
> 				bindKeys(Control.DOWN, [W, FlxKey.DOWN]);
> 				bindKeys(Control.LEFT, [Q, FlxKey.LEFT]);
> 				bindKeys(Control.RIGHT, [R, FlxKey.RIGHT]);
> 				bindKeys(Control.ACCEPT, [Z, SPACE, ENTER]);
> 				bindKeys(Control.BACK, [BACKSPACE, ESCAPE]);
> 				bindKeys(Control.PAUSE, [P, ENTER, ESCAPE]);
> 				bindKeys(Control.RESET, [T]);
> 		}
> ```

then you have to compile the code and run everything and idek, if you're on windows you can always use AutoHotKey
